predicate/src/rpc_predicate.rs

```diff
@@ -166,10 +190,52 @@ fn normalize_predicate(table_name: &str, predicate: &Predicate) -> Predicate {
             None => predicate.field_columns = Some(field_projections),
         };
     }
-
     predicate
 }
 
+struct SimplifyAdapter {
+    // TODO avoid re-creating this each time....
```

Comment:
The way to do so would be to cache the structure in the catalog. I will do so if we detect this taking too long during planning

Reply:
Or make a better way in DataFusion: https://github.com/apache/arrow-datafusion/issues/1725

```suggestion
    // TODO avoid re-creating this each time....
    // https://github.com/apache/arrow-datafusion/issues/1725
```

Reply:
https://github.com/apache/arrow-datafusion/pull/1726